src/tools/ci/test_extensions

```diff
@@ -0,0 +1,300 @@
+        ],
+        'build': [
+            ['make'],
+            ['make', 'install'],
+        ],
+        'test': [
+            # there is no test command,
+            # see bottom of https://github.com/pgpartman/pg_partman
+        ],
+    }
+)
+
+# add timescaledb
+# doesn't support PG 15 yet
+# extensions.append(
+#     {
+#         'name': 'timescaledb',
+#         'shared_preload_name': 'timescaledb',
+#         'download': [
+#             ['git', 'clone', 'https://github.com/timescale/timescaledb.git'],
+#         ],
+#         'build': [
+#             ['./bootstrap'],
+#             ['make', '-C', 'build'],
+#             ['make', '-C', 'build', 'install'],
+#         ],
+#         'test': [
+#             ['make', 'check'],
+#         ],
+#     }
+# )
+
+# add pgbouncer
+# since autoconf build is not working too, commented out for now
+# extensions.append(
+#     {
+#         'name': 'pgbouncer',
+#         'download': [
+#             git_clone_command
+#                 + ['https://github.com/pgbouncer/pgbouncer.git'],
+#         ],
+#         'build': [
+#             ['git', 'submodule', 'init'],
+#             ['git', 'submodule', 'update'],
+#             ['./autogen.sh'],
+#             ['./configure', '--prefix=/usr/local'],
+#             ['make'],
+#             ['make', 'install'],
+#         ],
+#         'test': [
+#             ['make', '-C', './test', 'all'],
+#             ['make', '-C', './test', 'check'],
+#         ],
+#     }
+# )
+
+# filter elements of only-run-ext list
+if only_run_ext:
+    # check if all elements of only-list exists
+    ext_names = [elem['name'] for elem in extensions]
+    if not set(only_run_ext).issubset(ext_names):
+        sys.exit('There are unsupported extensions. Supported extensions are: '
+                 + ' '.join(ext_names))
+
+    extensions = [ext for ext in extensions if ext['name'] in only_run_ext]
+
+
+def stop_and_start_postgres_server(logfile_prefix='',
+                                   shared_preload_string=''):
+    pgsql_data_path = '/tmp/pgsql/data'
+    logfile_name = '{}-logfile'.format(logfile_prefix).strip('-')
+
+    postgres_server_commands = [
+        ['pg_ctl', 'stop', '-D', pgsql_data_path],
+        ['rm', '-rf', pgsql_data_path],
+        ['mkdir', '-p', pgsql_data_path],
+        ['initdb', '-D', pgsql_data_path],
+        ['pg_ctl', '-D', pgsql_data_path, '-l', logfile_name, 'start'],
+    ]
+
+    if shared_preload_string:
+        shared_prelod_command = ['sed', '-i',
+                                 's/#shared_preload_libraries = ' +
+                                 '\'\'/shared_preload_libraries = \'' +
+                                 shared_preload_string + '\'/g',
+                                 pgsql_data_path + '/postgresql.conf']
+
+        postgres_server_commands.insert(len(postgres_server_commands) - 1,
+                                        shared_prelod_command)
```

Comment:
I'd probably just append this to the end of the config file. That's a lot easier. And perhaps this should allow appending arbitrary config contents?